### ISSUE TITLE
Start upload on submit ?

### ISSUE THREAD

**Issue by jtrollia** [danialfarid/ng-file-upload#169]
Hello,

Thanks for your plug.

I read https://github.com/danialfarid/angular-file-upload/issues/61 but I don't understand your explanation and don't see related issue in readMe. I tried many things but it seems var file can't be null or empty.
May I ask you to give few lines of code to illustrate your answer ?

Thanks again.

**Comment by bzitzow** [comment]
FWIW: I received this error when running jasmine specs via karma setting an expectation on what Upload.upload() should receive. 

The cause: 
```javascript
$scope.files = ['Im a pretend file for specs'];
```

The fix:
```javascript
var testFile = new Blob(['Im a pretend file for specs']);
$scope.files = [testFile];
```
